bin/syncutils/template.py

```diff
@@ -0,0 +1,86 @@
+import tempfile
+import utils
+import git
+import os
+import shutil
+from cookiecutter.main import cookiecutter
+
+class NfcoreTemplate:
+    """Updates the template content of an nf-core pipeline in
+    its `TEMPLATE` branch.
+
+    Args: - pipeline: The pipeline name
+          - branch: The template branch name, default=`TEMPLATE`
+          - token: GitHub auth token
+    """
+    def __init__(self, pipeline, branch='master', repo_url=""):
+        """Basic constructor
+        """
+        self.pipeline = pipeline
+        self.repo_url = repo_url
+        self.branch = branch
+        self.tmpdir = tempfile.mkdtemp()
+        self.templatedir = tempfile.mkdtemp()
+        self.repo = git.Repo.clone_from(self.repo_url, self.tmpdir)
+        assert self.repo
+    
+    def sync(self, template_url):
+        """Execute the template update.
+        """
+        context = self.context_from_nextflow(nf_project_dir=self.tmpdir)
+        self.update_child_template(template_url, self.templatedir, self.tmpdir, context=context)
+        self.commit_changes()
+        self.push_changes()
+
+    def context_from_nextflow(self, nf_project_dir):
+        """Fetch a Nextflow pipeline's config settings.
+
+        Returns: A cookiecutter-readable context (Python dictionary)
+        """
+        # Check if we are on "master" (main pipeline code)
+        if self.repo.active_branch is not "master":
+            self.repo.git.checkout("origin/master", b="master")
+
+        # Fetch the config variables from the Nextflow pipeline
+        config = utils.fetch_wf_config(wf_path=nf_project_dir)
+        
+        # Checkout again to configured template branch
+        self.repo.git.checkout("origin/{branch}".format(branch=self.branch),
+            b="{branch}".format(branch=self.branch))
+
+        return utils.create_context(config)
+    
+
+    def update_child_template(self, template_url, templatedir, target_dir, context=None):
+        """Apply the changes of the cookiecutter template
+        to the pipelines template branch.
+        """
+        cookiecutter(template_url,
```

Comment:
Any reason to do this directly using cookiecutter? It may be safer / more futureproof to use the `nf-core.create.run_cookiecutter` function instead, in case that is updated in the future... Especially if we add an output directory argument to that feature (which we should - I'll do that now).

Reply:
yeah, but how can I access this function from within my script?

Reply:
I mean, I can extend the system path so python will find the nf-core module, but it is not very consice